data-pipeline-flink/user-cache-updater-2.0/src/main/scala/org/sunbird/dp/usercache/util/UserMetadataUpdater.scala

```diff
@@ -0,0 +1,254 @@
+package org.sunbird.dp.usercache.util
+
+import java.util
+import java.util.Collections
+
+import com.datastax.driver.core.Row
+import com.datastax.driver.core.querybuilder.{Clause, QueryBuilder}
+import org.slf4j.LoggerFactory
+import org.sunbird.dp.core.cache.{DataCache, RedisConnect}
+import org.sunbird.dp.core.job.Metrics
+import org.sunbird.dp.core.util.CassandraUtil
+import org.sunbird.dp.usercache.domain.Event
+import org.sunbird.dp.usercache.task.UserCacheUpdaterConfigV2
+
+import scala.collection.JavaConverters.mapAsJavaMap
+import scala.collection.mutable
+
+class UserMetadataUpdater (config: UserCacheUpdaterConfigV2) {
+
+  private[this] val logger = LoggerFactory.getLogger(classOf[UserMetadataUpdater])
+  private val dataCache: DataCache = new DataCache(config, new RedisConnect(config.metaRedisHost, config.metaRedisPort, config), config.userStore, config.userFields)
+  private val cassandraConnect: CassandraUtil = new CassandraUtil(config.cassandraHost, config.cassandraPort)
```

Comment:
Should be done in ProcessFunction open() method and the connection closed in close() method

Reply:
Done